examples/evil-escape.el

```diff
@@ -0,0 +1,39 @@
+;;; init.el -*- lexical-binding: t; -*-
+
+;;; Commentary:
+
+;; An example Emacs Init file that extends `crafted-evil-packages'
+;; with `evil-escape', allowing alternative key combinations to escape
+;; from Vim's insert-mode.
+
+(setq custom-file (expand-file-name "custom.el" user-emacs-directory))
+(when (and custom-file
+	   (file-exists-p custom-file))
+  (load custom-file nil :nomessage))
+
+(load "~/crafted-emacs/modules/crafted-init-config")
+
+;;; Packages phase
+
+(require 'crafted-evil-packages)
+
+;; Add `evil-escape', the package responsible for binding Escape to an
+;; alternative key combination.
+(add-to-list 'package-selected-packages 'evil-escape)
+
+(package-install-selected-packages :noconfirm)
+
+;;; Configuration phase
+
+(require 'crafted-evil-config)
+
+;; Configure `evil-escape' with the preferred key combination "jj".
+(setq evil-escape-key-sequence "jj"
+      ;; Allow typing "jj" literally without exiting from insert-mode
+      ;; if the keys are pressed 0.2s apart.
+      evil-escape-delay 0.2
+      ;; Prevent "jj" from escaping any mode other than insert-mode.
+      evil-escape-inhibit-functions (list
+				     (lambda () (not (evil-insert-state-p)))))
```

Comment:
Not a request for change, more a point of discussion.

We (try to) follow a general [code style for Emacs](https://github.com/bbatsov/emacs-lisp-style-guide#functions) (see also #280).
Given that, this should probably be something like:
```emacs-lisp
(defun my/not-evil-insert-state-p ()
  "Inverse of `evil-insert-state-p`"
  (not (evil-insert-state-p)))

(customize-set-variable 'evil-escape-inhibit-functions (list #'my/not-evil-insert-state-p))
```

@jeffbowman thoughts?

Reply:
Agree.  

But the name should probably be `crafted-evil-not-insert-state-p`  rather than `my/...`.

Reply:
Makes sense, I used `my/...` as it's an example (therefor doesn't belong in the `crafted-evil` module), but also makes sense to not use `my/` for stuff we wrote.

Reply:
Oh... I thought the point was to have that in `crafted-evil`!  Yeah, if it is supposed to be in the user's config, `my/...` is perfectly fine.

Reply:
Yeah the intention is for this to be user-land, I'd imagine it's a frequent change but better as an example than something crafted-emacs would handle imo